cmd/minikube/cmd/start.go

```diff
@@ -1042,3 +1034,30 @@ func getKubernetesVersion(old *config.ClusterConfig) string {
 	}
 	return nv
 }
+
+// maybeExitWithAdvice before exiting will try to check for different error types and provide advice
+func maybeExitWithAdvice(err error) {
+	if errors.Is(err, oci.ErrWindowsContainers) {
+		out.ErrLn("")
+		out.ErrT(out.Conflict, "Your Docker Desktop container os type is Windows but Linux is required.")
+		out.T(out.Warning, "Please change Docker settings to use Linux containers instead of Windows containers.")
+		out.T(out.Documentation, "https://minikube.sigs.k8s.io/docs/drivers/docker/#verify-docker-container-type-is-linux")
+		exit.UsageT(`You can verify your Docker container type by running:
+{{.command}}
+	`, out.V{"command": "docker info --format '{{.OSType}}'"})
+	}
+
+	if errors.Is(err, oci.ErrCPUCountLimit) {
+		out.ErrLn("")
+		out.ErrT(out.Conflict, "Your {{.name}} doesn't have enough CPUs. ", out.V{"name": viper.GetString("driver")})
+		if runtime.GOOS != "linux" && viper.GetString("driver") == "docker" {
+			out.T(out.Warning, "Please consider changing your Docker desktop's resources.")
+			out.T(out.Documentation, "https://docs.docker.com/config/containers/resource_constraints/")
+		} else {
+			out.T(out.Warning, "Please ensure your system has at least {{.cpu_counts}} CPU cores", out.V{"cpu_counts": viper.GetInt(cpus)})
```

Comment:
If a user passes in `--cpus 1`, won't viper.GetInt() return 1?

We probably want to hardcode 2 in here, or have a `MinimumCPUs` variable

Reply:
for cpu 1 we already check for it in the validation,  so they user can not set it to 1 in first place
```
$ ./out/minikube start --driver=docker --cpus 1
😄  minikube v1.11.0 on Darwin 10.15.5
✨  Using the docker driver based on user configuration
💡  Requested cpu count 1 is less than the minimum allowed of 2
```